pkg/storage/ingest/reader.go

```diff
@@ -375,6 +377,10 @@ func (r *PartitionReader) processNextFetches(ctx context.Context, delayObserver
 	r.logFetchErrors(fetches)
 	fetches = filterOutErrFetches(fetches)
 
+	// instrument only after we've done all pre-processing of records. We don't expect the set of records to change beyond this point.
+	instrumentGaps(findGapsInRecords(fetches, r.lastSeenOffset), r.metrics.missedRecords, r.logger)
```

Comment:
I noticed noticed that we double log in `instrumentGaps()`. The 2nd log should be removed:

```diff
- level.Error(logger).Log("msg", "found gap in records", "start", gap.start, "end", gap.end)
```

Reply:
😕 i have no idea how we ended up here; removed